File: app/src/main/java/ru/melod1n/vk/database/MemoryCache.java
package ru.melod1n.vk.database;

import android.util.SparseArray;

import java.util.ArrayList;

import ru.melod1n.vk.api.model.VKGroup;
import ru.melod1n.vk.api.model.VKUser;

public class MemoryCache {

    private static SparseArray<VKUser> users = new SparseArray<>(30);
    private static SparseArray<VKGroup> groups = new SparseArray<>(30);

    public static VKUser getUser(int id) {
        VKUser user = users.get(id);
        if (user == null) {
            user = CacheStorage.getUser(id);
            if (user != null) {
                append(user);
            }
        }
        return user;
    }

    public static VKGroup getGroup(int id) {
        VKGroup group = groups.get(id);
        if (group == null) {
            group = CacheStorage.getGroup(id);
            if (group != null) {
                append(group);
            }
        }
        return group;
    }

    public static void update(ArrayList<VKUser> users) {
        for (VKUser user : users) {
            append(user);
        }
    }

    public static void append(VKGroup value) {
        groups.append(value.getId(), value);
    }

    public static void append(VKUser value) {
        users.append(value.getId(), value);
    }

    public static void clear() {
        users.clear();
        groups.clear();
    }

}

File: app/src/main/java/ru/melod1n/vk/database/MemoryCache.kt
package ru.melod1n.vk.database

import android.util.SparseArray
import ru.melod1n.vk.api.model.VKGroup
import ru.melod1n.vk.api.model.VKUser
import java.util.*

object MemoryCache {
    private val users = SparseArray<VKUser>(30)
    private val groups = SparseArray<VKGroup>(30)

    fun getUser(id: Int): VKUser? {
        var user = users[id]
        if (user == null) {
            user = CacheStorage.getUser(id)

            user?.let { append(it) }
        }
        return user
    }

    @JvmStatic
    fun getGroup(id: Int): VKGroup? {
        var group = groups[id]
        if (group == null) {
            group = CacheStorage.getGroup(id)

            group?.let { append(it) }
        }
        return group
    }

    fun update(users: ArrayList<VKUser>) {
        for (user in users) {
            append(user)
        }
    }

    fun append(value: VKGroup) {
        groups.append(value.id, value)
    }

    fun append(value: VKUser) {
        users.append(value.id, value)
    }

    fun clear() {
        users.clear()
        groups.clear()
    }
}